### PR TITLE
fix invalid runafter breaking pipeline builder issue

### PR DESCRIPTION
**Pull request by karthikjeeyar**
**Fixes**: 

https://issues.redhat.com/browse/ODC-5991

**Analysis / Root cause**: 

In yaml view, adding an non existent task name as a value to any task's `runafter` field and switching back to pipeline builder view forms an incorrect edge and topology layout fails to find the non existent node in the data model. 

**Solution Description**: 

Filter out the invalid/non-existent node name from runAfter field and also surface the invalid runAfter error to the user so they can correct their mistake by going back to yaml view.

**Screen shots / Gifs for design review**: 

Invalid_runAfter

**Unit test coverage report**: 

```
 getEdgesFromNodes
    ✓ should contain correct number of edges based on the runAfters (1ms)
    ✓ should exclude the invalid runAfter names
 ```

**Test setup:**

1. In create pipelines page, switch to pipeline builder
2. Try out `docker-build-and-deploy-pipeline` sample and change the name the `fetch-reposistory` task **or** add any invalid in `build` tasks's runAfter value
3. Switch back to pipeline builder view

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

cc: @bgliwa01 
/assign @andrewballantyne

### CODE REVIEW

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/const.ts [hunk 1]
```diff
@@ -18,13 +18,15 @@ export enum TaskErrorType {
   MISSING_RESOURCES = 'missingResources',
   MISSING_WORKSPACES = 'missingWorkspaces',
   MISSING_REQUIRED_WHEN_EXPRESSIONS = 'missingWhenExpressions',
+  INVALID_RUNAFTER = 'invalidRunAfter',
```
[the remark this answers is not in the document]
Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/const.ts:21]: Yes, Agreed. A `Message` and an `Action` is needed here to proceed with submission of the pipeline builder, Until this error is fixed the submit button stays disabled.

**Comment by karthikjeeyar** [comment]
@rohitkrai03 @andrewballantyne 
In this [commit](4a27196), I have added the form validation before switching to form view using `sanitizeTo` function in syncedEditorField. I have used the existing pipeline form schema to validate the errors. There could be some form errors which may not be necessary for yaml content eg `Must define at least one task` is a form validation error, so we should not show alert in this case, so we need to have a separate yaml schema(not as strict as form view) cc: @nemesis09 


invalid-form-or-invalid-structure

**Comment by nemesis09** [comment]
> @rohitkrai03 @andrewballantyne
> In this [commit](https://github.com/openshift/console/pull/9293/commits/4a27196ade84de03522998b3230b1712a32fa928), I have added the form validation before switching to form view using `sanitizeTo` function in syncedEditorField. I have used the existing pipeline form schema to validate the errors. There could be some form errors which may not be necessary for yaml content eg `Must define at least one task` is a form validation error, so we should not show alert in this case, so we need to have a separate yaml schema(not as strict as form view) cc: @nemesis09
> 
> invalid-form-or-invalid-structure

Thanks! @karthikjeeyar 
I agree, we should allow switch with any invalid value that can be corrected in the form. Maybe we can add a separate schema. I believe we have had similar discussion around this issue of invalid yaml switch requiring a second minimalized validation schema.
I have a implemented a similar schema as a part of https://github.com/openshift/console/pull/9282. I am still testing and refining it.
This PR already addresses much of the issue. Maybe I can update my PR to just add the min validation schema.

**Comment by andrewballantyne** [comment]
/close 

Will be included in #9282

**Pull request closed without merging**